src/control/lib/netdetect/netdetect.go

```diff
@@ -724,15 +744,39 @@ func convertLibFabricToMercury(provider string) (string, error) {
 // ValidateProviderStub is used for most unit testing to replace ValidateProviderConfig because the network configuration
 // validation depends upon physical hardware resources and configuration on the target machine
 // that are either not known or static in the test environment
-func ValidateProviderStub(provider string, device string) error {
+func ValidateProviderStub(device string, provider string) error {
 	// Call the full function to get the results without generating any hard errors
-	err := ValidateProviderConfig(provider, device)
+	log.Debugf("Calling ValidateProviderConfig with %s, %s", device, provider)
+	err := ValidateProviderConfig(device, provider)
 	if err != nil {
 		log.Debugf("ValidateProviderConfig (device: %s, provider %s) returned error: %v", device, provider, err)
 	}
 	return nil
 }
 
+// deviceProviderMatch is a helper function to consolidate this functionality used by the provider validation
+func deviceProviderMatch(deviceScanCfg DeviceScan, provider string, systemDevice string) bool {
+	deviceAffinity, err := GetAffinityForDevice(deviceScanCfg)
+	if err != nil {
+		return false
+	}
+	if deviceAffinity.DeviceName == systemDevice {
+		log.Debugf("Device %s supports provider: %s", systemDevice, provider)
+		return true
+	}
+	return false
+}
+
+func getHFIDeviceCount(hwlocDeviceNames []string) int {
+	var hfiDeviceCount int
+	for _, device := range hwlocDeviceNames {
+		if strings.Contains(device, "hfi1_") {
```

Comment:
would HasPrefix be more accurate here?

Reply:
Yes, HasPrefix would be more accurate.  I didn't know HasPrefix existed.  Do you prefer that I change it if there are other items to correct?